zend_abstract_interface/zai_string/string.h

```diff
@@ -5,28 +5,82 @@
 #include <stddef.h>
 #include <string.h>
 
+#include <Zend/zend.h>
+
+/**
+ * Represents a non-owning view of a string.
+ *
+ * When initializing the struct, use one of the initialization macros or
+ * functions. Do not initialize the struct directly e.g. `{0, null}`.
+ *
+ * todo: move .ptr to come before .len to match ddtrace_string and Rust and
+ *       ensure it's never null.
```

Comment:
Technically, an `Option<zai_string_view>` could be represented by ptr being null.